From 1ceeecebf092491d8c61dbd499a79a0eeee76da3 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 14 Mar 2024 08:38:47 +0200
Subject: [PATCH] [FSW-402] fs1: change register base defines

This patch changes register base define prefix from IO0_HALF0
to IO0_H0. This will shorten overall register definitions.

Change-Id: Idd1812fab892cb32126f9e8de97be29102bb1ccd
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 6ff7509..38a4cc0 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -19,7 +19,7 @@
 
 uint8_t boot_rom_get_die_id(void)
 {
-	uint32_t die_cpu_id = base_readl(IO0_HALF0_PSOC_ARC_MNG_LOCAL(mmARC_LOCAL_CONF_DIE_CPU_ID));
+	uint32_t die_cpu_id = base_readl(IO0_H0_PSOC_ARC_MNG_LOCAL(mmARC_LOCAL_CONF_DIE_CPU_ID));
 
 	return ((die_cpu_id & ARC_LOCAL_CONF_DIE_CPU_ID_DIE_ID_M)
 		>> ARC_LOCAL_CONF_DIE_CPU_ID_DIE_ID_S);
@@ -50,7 +50,7 @@
 	struct fs1_cold_rst_data cold_rst_data;
 	uint32_t i;
 	uint32_t die_cold_rst_flop_2 = HL_CFG_BASE +
-				       IO0_HALF0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2);
+				       IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2);
 
 	/* Replace correct die id in register offset*/
 	die_cold_rst_flop_2 = (die_cold_rst_flop_2 & ~(HL_ARC_ADDR_DIE_ID_MASK)) |
@@ -75,16 +75,16 @@
 void boot_rom_clear_status(void)
 {
 	/* As per Architecture recommendation, clear status reg */
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_STATUS));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_STATUS));
 
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_0));
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_1));
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_2));
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_3));
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_4));
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_5));
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_6));
-	regmap_writel(0, IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_7));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_0));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_1));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_2));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_3));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_4));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_5));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_6));
+	regmap_writel(0, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_7));
 
 	/* Die indication register is on io die slave and compute */
 	if (boot_rom_global.die_type != IO_DIE_MASTER)
@@ -93,14 +93,15 @@
 
 uint32_t boot_rom_get_status_base(void)
 {
-	return IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_0);
+	return IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_0);
 }
 
 void boot_rom_read_cold_flops(void)
 {
 	struct fs1_cold_rst_data cold_rst_data;
 
-	cold_rst_data.data = regmap_readl(IO0_HALF0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
+	cold_rst_data.data = regmap_readl(IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
+
 	boot_rom_global.recovery_flag = cold_rst_data.recovery_flag;
 	boot_rom_global.validation_pass = cold_rst_data.validation_flag;
 	boot_rom_global.efuse_read = cold_rst_data.efuse_read_flag;
@@ -122,7 +123,7 @@
 		return;
 
 	/* Perform arc1 iccm init only on cold reset */
-	cold_rst_data.data = regmap_readl(IO0_HALF0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
+	cold_rst_data.data = regmap_readl(IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
 	if (cold_rst_data.arc1_iccm_init_done)
 		return;
 
diff --git a/arch/fs1/arc/dev_delay.c b/arch/fs1/arc/dev_delay.c
index b265f6e..25e4b6c 100644
--- a/arch/fs1/arc/dev_delay.c
+++ b/arch/fs1/arc/dev_delay.c
@@ -14,14 +14,14 @@
 
 void boot_rom_clear_ts_counter(void)
 {
-	regmap_writel(0x0, IO0_HALF0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCR));
+	regmap_writel(0x0, IO0_H0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCR));
 }
 
 void boot_rom_start_timer(uint32_t timestamp_freq)
 {
-	regmap_writel(timestamp_freq, IO0_HALF0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTFID0));
+	regmap_writel(timestamp_freq, IO0_H0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTFID0));
 	regmap_writel((TIMESTAMP_CNTCR_CNTCR_EN_M | TIMESTAMP_CNTCR_CNTCR_HDBG_M),
-		      IO0_HALF0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCR)); /* EN | HDBG */
+		      IO0_H0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCR)); /* EN | HDBG */
 }
 
 uint64_t boot_rom_get_timestamp_freq(void)
@@ -34,12 +34,12 @@
 {
 	uint32_t low, up1, up2;
 
-	up1 = regmap_readl(IO0_HALF0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVU));
-	low = regmap_readl(IO0_HALF0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVL));
-	up2 = regmap_readl(IO0_HALF0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVU));
+	up1 = regmap_readl(IO0_H0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVU));
+	low = regmap_readl(IO0_H0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVL));
+	up2 = regmap_readl(IO0_H0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVU));
 	/* In case of overlap read low part again */
 	if (up1 != up2)
-		low = regmap_readl(IO0_HALF0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVL));
+		low = regmap_readl(IO0_H0_PSOC_TIMESTAMP(mmTIMESTAMP_CNTCVL));
 
 	return ((uint64_t)up2 << 32) + low;
 }
diff --git a/arch/fs1/arc/dev_spi_flash.c b/arch/fs1/arc/dev_spi_flash.c
index 2f52ebf..f56be1e 100644
--- a/arch/fs1/arc/dev_spi_flash.c
+++ b/arch/fs1/arc/dev_spi_flash.c
@@ -19,7 +19,7 @@
 
 	/* clear BYTE SWAP in SPI DMA so we dont switch endianness */
 	reg_spi_direct_wr_rd_ctrl =
-		regmap_readl(IO0_HALF0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL));
+		regmap_readl(IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL));
 	if (enable)
 		reg_spi_direct_wr_rd_ctrl |=
 			GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_M;
@@ -28,15 +28,15 @@
 			~GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_M;
 
 	regmap_writel(reg_spi_direct_wr_rd_ctrl,
-		      IO0_HALF0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL));
+		      IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL));
 }
 
 void spi_flash_set_baud_rate(uint8_t baud_rate)
 {
-	regmap_writel(baud_rate, IO0_HALF0_PSOC_QSPI_BASE(mmSPI_BAUDR));
+	regmap_writel(baud_rate, IO0_H0_PSOC_QSPI(mmSPI_BAUDR));
 }
 
 void spi_flash_set_write_limit(enum flash_write_limit write_limit)
 {
-	regmap_writel(write_limit, IO0_HALF0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_SPI_WRITE_LIMIT));
+	regmap_writel(write_limit, IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_SPI_WRITE_LIMIT));
 }
